org.civicrm.reqangular/src/common/decorators/q/q-sequence.js

```diff
@@ -0,0 +1,23 @@
+/* eslint-env amd */
+
+define([], function () {
+  return ['$delegate', function ($delegate) {
+    $delegate.sequence = executePromisesInSequence;
+
+    /**
+     * Executes an array of promises sequentially
+     *
+     * @param {Array} promises collection of promises
+     * @return {Promise}
+     */
+    function executePromisesInSequence (promises) {
+      var sequence = promises.reduce(function (promiseChain, promise) {
+        return promiseChain.then(promise);
+      }, $delegate.resolve());
+
+      return $delegate.all(sequence);
```

Comment:
Doesn't `promises.reduce` return a single promise value instead of an array? I don't think $delegate.all is needed here (at least in this scenario)